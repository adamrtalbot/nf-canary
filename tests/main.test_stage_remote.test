nextflow_process {

    name "Test Process TEST_STAGE_REMOTE"
    script "main.nf"
    process "TEST_STAGE_REMOTE"

    test("Should run without failures") {

        when {

            params {
              remoteFile = "https://raw.githubusercontent.com/adamrtalbot/nf-canary/main/README.md"
            }

            process {
                """
                input[0] = Channel.fromPath(params.remoteFile)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
