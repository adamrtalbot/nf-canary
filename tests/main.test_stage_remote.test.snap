{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "[![nf-test](https://github.com/adamrtalbot/nf-canary/actions/workflows/nf-test.yml/badge.svg)](https://github.com/adamrtalbot/nf-canary/actions/workflows/nf-test.yml)\n\n# :bird: nf-canary\n\nA minimal Nextflow workflow for testing infrastructure.\n\n## Introduction\n\nAfter setting up and deploying your infrastructure, you may need to test you can effectively run a Nextflow pipeline. Common issues include network problems, permissions and invalid paths. The aim of this pipeline is to run through a set of standard tasks and confirm they are able to work. By using simple tasks which test as few things as possible, we can isolate individual any problems that may be occurring and fix them.\n\nThis pipeline aims to be as simple as possible. Therefore it comes with no additional configuration and the user will need to configure the run for their own infrastructure.\n\nAll tests are in the `main.nf` file and aim to be as simple and legible as possible. If it is difficult to understand a test, please raise an issue.\n\n## Usage\n\n### Running locally\n\nTo run locally, you can simply run:\n\n```\nnextflow run adamrtalbot/nf-canary\n```\n\nThis will execute on your local machine with default settings. All tests should pass here. If they do not something is incorrect about your Nextflow installation or configuration.\n\n### Running on your infrastructure\n\nAdd the configuration for your infrastructure using the relevant configuration files as listed in [the Nextflow documentation](https://www.nextflow.io/docs/latest/config.html). You can then run the same pipeline but with the additional configuration to run on your set up.\n\n### Skipping Tests\n\nEach test can be skipped by name with the parameter `--skip`, e.g. `--skip TEST_INPUT`. Multiple can be specified with comma delimited values, e.g. `--skip TEST_CREATE_FILE,TEST_PASS_FILE`. Note, some tests depend on previous tests running so will be skipped if an upstream test does not run (or fails). All tests which are dependent on previous tests are listed here:\n\n```yaml\nTEST_CREATE_FILE:\n  - TEST_PASS_FILE\n```\n\n### Interpreting the results\n\nThe code for each test includes a short comment explaining what it is aiming to test. If it fails, check the comment for what it was trying to achieve and compare that to the error message reported by Nextflow.\n"
                ]
            }
        ],
        "timestamp": "2023-07-20T15:24:52+0000"
    }
}